### PR TITLE
PLATFORM-3800 | Add language wikis index and improve handling of closed wikis

### PR DESCRIPTION
**Pull request by rogatty**
## Links

* http://wikia-inc.atlassian.net/browse/PLATFORM-3800
* https://github.com/Wikia/app/pull/16374

## Description

- When a wiki is closed, and there are no language wikis or user loads a path other than `/language-wikis` then show a page with message that the wikis is closed with status 410. Previously, user would see a desktop page.
- When a wiki is closed, there are language wikis and user loads the `/language-wikis` path then show a page with a list of language wikis.
- When a wiki doesn't exist, there are language wikis and user loads the `/language-wikis` path then show a page with a list of language wikis.
- When a wiki doesn't exist, there are language wikis and user loads a path other than `/language-wikis` then redirect to `/language-wikis`.

## Reviewers

@Wikia/core-platform-team @Wikia/iwing

### CODE REVIEW

**Comment by themech** [comment]
Assets seem to be exceeding current size limits

File: app/router.js [hunk 1]
```diff
@@ -20,9 +20,15 @@ const Router = EmberRouter.extend(RouterScroll, {
 /* eslint array-callback-return: 0 */
 Router.map(function () {
   this.route('article-preview');
-
+  this.route('closed-wiki', {
+    path: '/__closed_wiki__this_is_never_visible_to_user',
```
Comment (rybmat) [inline app/router.js:24]: nice path ;)

File: package.json [hunk 2]
```diff
@@ -41,7 +41,7 @@
     "compression": "1.7.3",
     "cors": "2.8.4",
     "deep-extend": "0.6.0",
-    "design-system": "github:Wikia/design-system#15.3.4",
+    "design-system": "github:Wikia/design-system#29b766fbabac3a3bf2e2758b37aa4456f80e3950",
```
Comment (rybmat) [inline package.json:44]: remember to release DS and use release tag here

**Review by rybmat — APPROVED** [review]

File: app/routes/language-wikis-index.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+import HeadTagsDynamicMixin from '../mixins/head-tags-dynamic';
+
+export default Route.extend(
+  HeadTagsDynamicMixin,
+  {
+    wikiVariables: service(),
+
+    beforeModel() {
+      if (!this.wikiVariables.languageWikis) {
```
Comment (themech) [inline app/routes/language-wikis-index.js:14]:
```suggestion
      // Product requirement - on open wikis the /language-wikis path should redirect to the main page
      if (!this.wikiVariables.languageWikis) {
```

**Review by themech — APPROVED** [review]

File: app/router.js (hunk 1 above)
Comment (rogatty) [reply app/router.js:24]: Got the idea from https://github.com/emberjs/ember.js/blob/9830f066ae2ef4fad4bfa63f5cc43997d8ac466b/packages/%40ember/-internals/routing/lib/system/dsl.ts#L46 ;)

File: app/templates/language-wikis-index.hbs [hunk 4]
```diff
@@ -0,0 +1 @@
+{{language-wikis-index}}
```
Comment (hakubo) [inline app/templates/language-wikis-index.hbs]: can we render this directly here, instead of the component wrapper? 
It'll be easier to follow

File: app/templates/components/site-head.hbs [hunk 5]
```diff
@@ -19,8 +19,11 @@
...
 </div>
 {{wds-banner-notifications}}
-{{wds-community-bar
-  model=communityHeader
-  track=(action track)
-  onLinkClicked=(action onLinkClicked)
-}}
+{{! It's not available on language wikis index }}
+{{#if communityHeader}}
+  {{wds-community-bar
```
Comment (hakubo) [inline app/templates/components/site-head.hbs:24]: while at it would you mind:
```suggestion
  <WdsCommunityBar @model={{communityHeader}} @track={{action track}} onLinkClicked={{action onLinkClicked}} />
```

File: app/templates/closed-wiki.hbs [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<div class="wiki-container closed-wiki">
+  {{wikia-ui-components/wiki-page-header
+    title="This wiki has been closed or disabled"
+  }}
+  {{#wds-button
```
Comment (hakubo) [inline app/templates/closed-wiki.hbs:5]: <WdsButton @href="?useskin=oasis" @class="closed-wiki__button" />

would be nicer with splattributes but DS does not support it yet

File: app/templates/closed-wiki.hbs [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<div class="wiki-container closed-wiki">
+  {{wikia-ui-components/wiki-page-header
```
Comment (hakubo) [inline app/templates/closed-wiki.hbs:2]: Angle Brackets do not support such invocation yet, so don't try :)

File: app/styles/component/_language-wikis-index.scss [hunk 8]
```diff
@@ -0,0 +1,43 @@
+.language-wikis-index {
+  margin: 50px $article-horizontal-padding 0;
+  text-align: center;
+
+  &__intro {
+    font-size: $wds-font-size-l;
```
Comment (hakubo) [inline app/styles/component/_language-wikis-index.scss]: https://fandomdesignsystem.com/base-styles/typography - please use classes not to bloat CSS size

File: app/routes/language-wikis-index.js [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+export default Route.extend(
+  HeadTagsDynamicMixin,
+  {
+    wikiVariables: service(),
+
+    beforeModel() {
+      // Product requirement - on open wikis the /language-wikis path should redirect to the main page
+      if (!this.wikiVariables.languageWikis) {
+        this.fastboot.set('response.statusCode', 301);
```
Comment (hakubo) [inline app/routes/language-wikis-index.js]: this.set('fastboot.response.statusCode

File: app/mixins/closed-wiki-handler.js [hunk 10]
```diff
@@ -0,0 +1,17 @@
+import { inject as service } from '@ember/service';
+import Mixin from '@ember/object/mixin';
+
+import { WikiIsClosedError } from '../utils/errors';
+
+export default Mixin.create({
+  fastboot: service(),
```
Comment (hakubo) [inline app/mixins/closed-wiki-handler.js]: not used?

File: app/mixins/closed-wiki-handler.js [hunk 11]
```diff
@@ -0,0 +1,17 @@
+import { inject as service } from '@ember/service';
+import Mixin from '@ember/object/mixin';
+
+import { WikiIsClosedError } from '../utils/errors';
+
+export default Mixin.create({
```
Comment (hakubo) [inline app/mixins/closed-wiki-handler.js]: Mixins are hard to follow aaaand are a root of all evil :)

this one is so small that I'd suggest to copy and paste to where this logic is needed to make it simplier to understand

File: app/mixins/empty-domain-with-language-wikis-handler.js [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+import { DontLogMeError } from '@wikia/ember-fandom/utils/errors';
+
+export default Mixin.create({
+  fastboot: service(),
+  wikiVariables: service(),
+
+  beforeModel() {
+    this._super(...arguments);
+
+    if (
```
Comment (hakubo) [inline app/mixins/empty-domain-with-language-wikis-handler.js]: how about creating an util that would do this and you could just import it to where it is needed
```
handleEmptyDomainWithLanguageWikis(this.fastboot, this.wikiVariables);
```

File: app/templates/components/site-head.hbs (hunk 5 above)
Comment (rogatty) [reply app/templates/components/site-head.hbs:24]: I see that we don't use angle brackets anywhere in mobile-wiki yet. I'd prefer to not introduce them in this PR as it can have unknown side-effects and I don't want to leave the Platform team with a problem like this to fix. I suggest converting them all in a single, separate PR, instead.

File: app/utils/closed-wiki-handler.js [hunk 13]
```diff
@@ -0,0 +1,7 @@
+import { WikiIsClosedError } from './errors';
```
Comment (hakubo) [inline app/utils/closed-wiki-handler.js:1]: thank you!
BTW we'll have to enable this
https://github.com/ember-cli/eslint-plugin-ember/blob/HEAD/docs/rules/no-new-mixins.md :)

File: app/routes/wiki-page.js [hunk 14]
```diff
@@ -59,6 +62,9 @@ export default Route.extend(
     beforeModel(transition) {
       this._super(transition);
 
+      closedWikiHandler(this.wikiVariables);
```
Comment (hakubo) [inline app/routes/wiki-page.js:65]: so much better, thank you!

**Review by hakubo — APPROVED** [review]

File: app/routes/wiki-page.js (hunk 14 above)
Comment (rogatty) [reply app/routes/wiki-page.js:65]: It's a controller, it doesn't have `this.element`

**Review by themech — APPROVED** [review]
